connector/src/main/java/org/geysermc/connector/network/translators/java/entity/spawn/JavaSpawnPaintingTranslator.java

```diff
@@ -50,7 +51,11 @@ public void translate(ServerSpawnPaintingPacket packet, GeyserSession session) {
             )
                     .setPaintingName(PaintingType.getByPaintingType(packet.getPaintingType()))
                     .setDirection(packet.getDirection().ordinal());
-
+            try {
+                TimeUnit.MILLISECONDS.sleep(100);
```

Comment:
Instead of sleeping for 100 milliseconds, put `session.getConnector().getGeneralThreadPool().schedule(() -> session.getEntityCache().spawnEntity(entity), 100, TimeUnit.MILLISECONDS)`.